modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/StyleBookModal.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput, ClaySelect} from '@clayui/form';
+import ClayModal, {useModal} from '@clayui/modal';
+import {FieldBase} from 'frontend-js-components-web';
+import {fetch, navigate, objectToFormData} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useReducer, useRef, useState} from 'react';
+
+const DEFAULT_OPTION = {
+	label: `- ${Liferay.Language.get('not-selected')} -`,
+	value: '-1',
+};
+
+export default function StyleBookModal({
+	addStyleBookEntryURL,
+	frontendTokenDefinitions = [],
+	namespace,
+	onModalClose,
+}) {
+	const {observer, onClose} = useModal({onClose: () => onModalClose()});
+
+	const [loading, setLoading] = useState(false);
+
+	const [errors, setErrors] = useReducer(
+		(value, nextValue) => ({...value, ...nextValue}),
+		{frontendTokenDefinition: null, name: null}
+	);
+
+	const [name, setName] = useState('');
+	const [frontendTokenDefinition, setFrontendTokenDefinition] =
+		useState(null);
+
+	const formRef = useRef(null);
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+
+		const errors = validateFields(name, frontendTokenDefinition);
+
+		if (Object.keys(errors).length) {
+			setErrors(errors);
+
+			return;
+		}
+
+		setLoading(true);
+
+		const body = Liferay.Util.ns(namespace, {
+			name,
+			themeId: frontendTokenDefinition.themeId,
+		});
+
+		fetch(addStyleBookEntryURL, {
+			body: objectToFormData(body),
+			method: 'POST',
+		})
+			.then((response) => response.json())
+			.then((responseContent) => {
+				if (responseContent.error) {
+					setLoading(false);
+					setErrors(responseContent.error);
+				}
+				else if (responseContent.redirectURL) {
+					navigate(responseContent.redirectURL, {
+						beforeScreenFlip: onClose,
+					});
+				}
+			})
+			.catch(() =>
+				setErrors({
+					other: Liferay.Language.get(
+						'an-unexpected-error-occurred-while-creating-the-style-book'
+					),
+				})
+			);
+	};
+
+	const nameId = `${namespace}name`;
+	const frontendTokenDefinitionId = `${namespace}tokenDefinition`;
+
+	return (
+		<ClayModal observer={observer} size="md">
+			<ClayModal.Header>
+				{Liferay.Language.get('add-style-book')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				{errors.other && (
+					<ClayAlert
+						displayType="danger"
+						onClose={() => {}}
+						title={Liferay.Language.get('error')}
+					>
+						{errors.other}
+					</ClayAlert>
+				)}
+
+				<FieldBase
+					errorMessage={errors.frontendTokenDefinition}
+					id={frontendTokenDefinitionId}
+					label={Liferay.Language.get('create-style-book-for')}
+					name="frontendTokenDefinition"
+					required={true}
+				>
+					<ClaySelect
```

Comment:
We should be setting required here and on the `FieldBase` for both inputs. I'm unable to submit the form when one or both inputs are empty with no feedback on why it's not submitting.

Reply:
Are you testing with these changes added? For me when you submit with one or both inputs empty, a red message appears that says "This field is required"